### PR TITLE
respect Spree::Config setting in respect to use ‚state‘ in addresses

### PR DESCRIPTION
**Pull request by MobilUser**
align way to display or hide the state field in form with the method using is solidus core

### CODE REVIEW

**Comment by MobilUser** [comment]
forget test adaptations

**Comment by MobilUser** [comment]
I see 13 failures in test but none related to this simple fix here

File: lib/views/frontend/spree/checkout/payment/_paypal_braintree.html.erb [hunk 1]
```diff
@@ -22,7 +22,9 @@
       line1: '<%= address.address1 %>',
       line2: '<%= address.address2 %>',
       city: '<%= address.city %>',
-      state: '<%= address.state.name %>',
+      <% if Spree::Config[:address_requires_state] %>
+        state: '<%= address.state.name %>',
+      <% end %>
```
Comment (tvdeyen) [inline lib/views/frontend/spree/checkout/payment/_paypal_braintree.html.erb]: I think we should always transfer the state if we have one, on matter what this config is.

File: config/locales/de.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    nonce: Nonce
+    token: Token
+    payment_type:
+      apple_pay_card: Apple Pay
+      pay_pal_account: PayPal
+    configurations:
+      title: Braintree Einstellungen
+      tab: Braintree
+      update_success: Einstellungen gespeichert.
+      update_error: Es ist ein Fehler beim Speichern der Einstellungen aufgetreten.
```
Comment (tvdeyen) [inline config/locales/de.yml]: Could you please send this as a separate PR?

File: app/assets/javascripts/spree/frontend/solidus_paypal_braintree.js [hunk 3]
```diff
@@ -1,5 +1,9 @@
 //= require spree/braintree_hosted_form
 
+//= require ../braintree_3_5_0_client
+//= require ../braintree_3_5_0_paypal
+//= require ../braintree_3_5_0_data_collector
+
```
Comment (tvdeyen) [inline app/assets/javascripts/spree/frontend/solidus_paypal_braintree.js]: I don't think this belongs into this PR. We are loading the client lib from Braintree on demand and don't include this anymore.

**Comment by tvdeyen** [comment]
Closing for now. Please open separate PRs. Thanks for the contribution
